OpenTap.Metrics/AssetDiscovery/IAssetDiscovery.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+
+namespace OpenTap.Metrics.AssetDiscovery;
+
+/// <summary>
+/// Plugin interface for asset discovery.
+/// Implementations of this interface can be used to discover assets such as instruments or DUTs connected to the system.
+/// </summary>
+public interface IAssetDiscovery : OpenTap.ITapPlugin
+{
+    /// <summary>
+    /// Sets the priority of this provider. In case two implementations return the same 
+    /// discovered asset (same Identifier), the one from the higher priority provider is used
+    /// </summary>
+    double Priority { get; }
+
+    /// <summary>
+    /// Discovers assets connected to the system.
+    /// </summary>
+    IEnumerable<DiscoveredAsset> DiscoverAssets();
+}
+
+/// <summary>
+/// Represents a discovered asset.
+/// IAssetDiscovery implementations can specialize this class to provide additional information.
+/// </summary>
+public class DiscoveredAsset
+{
+    /// <summary>
+    /// The type of the asset. E.g. "N9020A".
+    /// This can be used to determine a suitable driver for the asset
+    /// so it can be used as an Asset in OpenTAP.
+    /// </summary>
+    public string Model { get; set; }
+    /// <summary>
+    /// A unique identifier for the asset. This is used to identify the asset in the system.
+    /// E.g. for an Instrument, this could be a combination of Manufacturer, Model and serial number.
+    /// </summary>
+    public string Identifier { get; set; }
+    /// <summary>
+    /// The manufacturer of the asset. E.g. "Keysight". 
+    /// </summary>
+    public string Manufacturer { get; set; }
```

Comment:
Should we add this here?

```
public AssetStatus Status { get; set; }

...

enum AssetStatus
{
    Connected,
    Disconnected,
    InUse
}

Reply:
I was thinking the same thing, but I didn't do it because I don't think we can expect a IAssetDiscovery service (in the idle session) to know when something is InUse (in a different session).

Reply:
What about something like:

```cs
[Flags]
enum AssetStatus 
{
     Lost = 0,  // It was visible once but is no longer (optional).
     InSettings = 1, // Its defined in bench settings, but not currently seen.
     Visible = 2, // Its visible, meaning it shows up during active discovery.
     InUse = 4 // Its currently in use.
}

Reply:
* I'm not sure we should ask IAssetDiscovery plugins to determine if an asset is in use or not.
* Lost implies state now needs to be held in the implementation between calls to Discover. 
* I don't think we have the model and serial number of something that is only defined in bench settings. I see the entry in bench settings as a "slot" for a resource (E.g. for a DUT), the actual resource is only determined after connecting (e.g. its serial number/mac address/die id). At least opentap is sometimes used like that.

Reply:
I think that is fine. We can anyway extend the class at a later point.